src/Umbraco.Web/UmbracoInjectedModule.cs

```diff
@@ -99,9 +83,30 @@ private void BeginRequest(HttpContextBase httpContext)
             // ensure there's an UmbracoContext registered for the current request
             // registers the context reference so its disposed at end of request
             var umbracoContextReference = _umbracoContextFactory.EnsureUmbracoContext(httpContext);
+            umbracoContextReference.UmbracoContext.CreatingPublishedSnapshot += UmbracoContext_CreatingPublishedSnapshot;
```

Comment:
do we need to do all this even when not purelive?

Reply:
Not necessarily, when not in pure live we don't need to wait and any request that might execute while notifying is happening on a background thread will just serve the previous generation which is OK. I'll think about how we can encapsulate this stuff without having to have PureLive checks everywhere.

Reply:
Have pushed a change for this, though it rudimentary (i.e. not the prettiest) but it will do,